lib/workers/repository/process/write.ts

```diff
@@ -50,6 +72,46 @@ export async function writeUpdates(
       incLimitedValue(Limit.Branches);
     }
   }
+  const rollupBranches: BranchConfig[] = [];
+  for (const [rollupBranchName, statuses] of Object.entries(rollups)) {
+    let branch: BranchConfig;
+    if (statuses[BranchStatus.green]?.length) {
+      branch = statuses[BranchStatus.green].pop();
+      logger.debug(
+        `Adding ${branch.upgrades.length} upgrades to rollup branch ${rollupBranchName} from ${branch.branchName}`
+      );
+      statuses[BranchStatus.green].forEach((greenBranch) => {
+        logger.debug(
+          `Adding ${greenBranch.upgrades.length} upgrades to rollup branch ${rollupBranchName} from ${greenBranch.branchName}`
+        );
+        branch.upgrades = branch.upgrades.concat(greenBranch.upgrades);
+      });
+    } else if (statuses[BranchStatus.yellow]?.length) {
+      logger.debug(`Rollup branch ${rollupBranchName} has no passing branches`);
+      branch = statuses[BranchStatus.yellow].pop();
+      delete branch.upgrades;
+      // TODO: PR body
+    } else if (statuses[BranchStatus.red]?.length) {
+      logger.debug(
+        `Rollup branch ${rollupBranchName} has only failing branches`
+      );
+    }
+    if (branch) {
+      branch.isRollup = true;
+      branch.branchName = rollupBranchName;
+      branch.prTitle = branch.rollupName; // TODO: templating?
```

Comment:
Does it apply semantic commit prefixes?

Reply:
Yeah, definitely needs to incorporate semantic commits

Reply:
Can we simply extract and reuse code from normal updates?